socialhome/streams/app/tests/components/AuthorBar.tests.js

```diff
@@ -0,0 +1,210 @@
+                    console.error.calledWith("Not logged in").should.be.true
+                    console.error.restore()
+                }))
+                target.instance().unfollow()
+            })
+        })
+
+        describe("follow", () => {
+            it("should display an error message if the user is not logged in", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: null})
+                let target = mount(AuthorBar, {propsData})
+                sinon.spy(console, "error")
+                target.instance().follow()
+                console.error.calledWith("Not logged in").should.be.true
+                console.error.restore()
+            })
+
+            it("should not send an HTTP request if the user is not logged in", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: null})
+                let target = mount(AuthorBar, {propsData})
+                sinon.spy(target.instance().$http, "post")
+                target.instance().follow()
+                target.instance().$http.post.called.should.be.false
+            })
+
+            it("should send an HTTP request with the right parameters when user is logged in", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: "26", guid: "42"})
+                let target = mount(AuthorBar, {propsData})
+                sinon.spy(target.instance().$http, "post")
+                target.instance().follow()
+                target.instance().$http.post
+                    .calledWith("/api/profiles/26/add_follower/", {guid: "42"}).should.be.true
+            })
+
+            it("should show that user is following author when the HTTP request succeeds", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: "26", guid: "42", isUserFollowingAuthor: false})
+                let target = mount(AuthorBar, {propsData})
+                sinon.stub(target.instance().$http, "post").returns(new Promise(resolve => {
+                    resolve()
+                    target.instance().$data._isUserFollowingAuthor.should.be.true
+                    target.find(".unfollow-btn").length.should.equal(1)
+                }))
+                target.instance().follow()
+            })
+
+            it("should show an error message when the HTTP request fails", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: "26", guid: "42", isUserFollowingAuthor: true})
+                let target = mount(AuthorBar, {propsData})
+                sinon.stub(target.instance().$http, "post").returns(new Promise(_, reject => {
+                    reject()
+                    sinon.spy(console, "error")
+                    console.error.calledWith("Not logged in").should.be.true
+                    console.error.restore()
+                }))
+                target.instance().follow()
+            })
+        })
+    })
+
+    describe("lifcycle", () => {
```

Comment:
![](https://diaspora-fr.org/camo/bb946dd1d6ce9ee3ad01d39c08b602b69003fd6a/68747470733a2f2f737461746963312e73717561726573706163652e636f6d2f7374617469632f3531386635643632653462303735323438643661336639302f742f3539393033623131636638316530323666363330376262322f313530323632343630363434382f3f666f726d61743d3130303077)

Though I'd like to keep the `lifecycle`. I think it's a nice convention for methods that belong to the component lifecycle. The same way `computed` and `methods` have their own describe. I think tests are more readable organized this way ;)